### DISCUSSION TITLE
Unauthorized websocket

### DISCUSSION THREAD

**Discussion by Korijn** [miguelgrinberg/python-engineio#236] [Q&A]
I've been trying to implement ConnectionRefused exception messaging to signal to my clients _why_ their connections are refused.

However, my clients use the socket.io library with the websocket transport only (polling transport disabled). And it seems that that triggers the following [codepath](https://github.com/miguelgrinberg/python-engineio/blob/main/src/engineio/async_drivers/asgi.py#L188):

```py
async def make_response(status, headers, payload, environ):
    headers = [(h[0].encode('utf-8'), h[1].encode('utf-8')) for h in headers]
    if environ['asgi.scope']['type'] == 'websocket':
        if status.startswith('200 '):
            await environ['asgi.send']({'type': 'websocket.accept',
                                        'headers': headers})
        else:
            await environ['asgi.send']({'type': 'websocket.close'})   # <---- codepath ends here
        return

    await environ['asgi.send']({'type': 'http.response.start',
                                'status': int(status.split(' ')[0]),
                                'headers': headers})
    await environ['asgi.send']({'type': 'http.response.body',
                                'body': payload})   # <---- this contains my connection refused message
```

And the payload is never returned to the client.

Does this mean that it only works with the socket.io polling transport?

**Comment by miguelgrinberg** [comment]
The [websocket.close](https://asgi.readthedocs.io/en/latest/specs/www.html#close-send-event) event in ASGI does not allow a payload to be included.

I'll have to think about this to see if there is any way within the ASGI and Socket.IO rules to pass on an error message with a rejection.

**Reply by miguelgrinberg** [reply]
I think I've got this fixed. Can I ask you to test it with the main branch of this repository? For your reference, the fix is here: https://github.com/miguelgrinberg/python-engineio/commit/583c7db1f9dfa62290481634b4f9ab4d39a8ec6b. Thanks!
